templates/default/origin-dns.conf.erb

```diff
@@ -1,5 +1,7 @@
-strict-order
 no-resolv
 domain-needed
-server=/cluster.local/<%=node['cookbook-openshift3']['openshift_common_first_svc_ip'] %>
```

Comment:
Without this line, all pod DNS queries are sent to local dnsmasq, which forwards them to the upstream. No queries are sent to the kubernetes DNS, thus breaking pods which depend on it (such as statefulset).

For example, this is the end of the log the first pod of my mysql statefulset, which uses the peer-finder script:
```
+ /usr/bin/peer-finder -on-start=/usr/share/container-scripts/mysql/configure-galera.sh -service=mysql
2017/10/12 09:31:26 lookup mysql on 10.0.2.15:53: no such host
2017/10/12 09:31:27 lookup mysql on 10.0.2.15:53: no such host
2017/10/12 09:31:28 lookup mysql on 10.0.2.15:53: no such host
2017/10/12 09:31:29 lookup mysql on 10.0.2.15:53: no such host
```

Doing the following fixes the issue in the short term:
```sh
echo 'server=/cluster.local/172.30.0.1' | tee -a /etc/dnsmasq.d/origin-dns.conf
service dnsmasq restart
```

I suggest that you restore this line :)

Reply:
I have also been testing and everything is ok for me :/ it ts strange...

Reply:
@IshentRas here is step by step instruction, using the kitchen tests (attention: I push --amend a small change to the test, please pull my changes before running):

```sh
host$ kitchen converge cluster-native-ose36-centos-72
host$ kitchen login cluster-native-ose36-centos-72
vm$ sudo -i # inside the VM
vm# oc adm uncordon origin-centos-72
vm# oc project default
# wait until router is running
vm# oc exec -ti router-2-1jwmw -- /bin/bash
[root@origin-centos-72 ~]# oc exec -ti router-2-1jwmw -- /bin/bash
Defaulting container name to router.
Use 'oc describe pod/router-2-1jwmw' to see all of the containers in this pod.
/usr/bin/id: cannot find name for user ID 1000040000
[I have no name!@origin-centos-72 conf]$ 
[I have no name!@origin-centos-72 conf]$ ping router.default.svc.cluster.local
ping: router.default.svc.cluster.local: Name or service not known
[I have no name!@origin-centos-72 conf]$ 
# open another "kitchen login" and edit /etc/dnsmasq.d/origin-dns.conf like this:
# `echo 'server=/cluster.local/172.30.0.1' | sudo tee -a /etc/dnsmasq.d/origin-dns.conf && sudo service dnsmasq restart`
# then go back to the router pod to retry the above ping test
[I have no name!@origin-centos-72 conf]$ ping router.default.svc.cluster.local
PING router.default.svc.cluster.local (172.30.206.236) 56(84) bytes of data.
--- router.default.svc.cluster.local ping statistics ---
2 packets transmitted, 0 received, 100% packet loss, time 1000ms
```